python/ray/data/_internal/execution/legacy_compat.py

```diff
@@ -200,13 +218,7 @@ def _blocks_to_input_buffer(blocks: BlockList, owns_blocks: bool) -> PhysicalOpe
                             # This isn't a proper block, but it's what we are doing
                             # in the legacy code.
                             ray.put(read_task),
-                            BlockMetadata(
-                                num_rows=1,
-                                size_bytes=len(cloudpickle.dumps(read_task)),
```

Comment:
This is the bug. It should `read_task.get_metadata().size_bytes` instead of the cloudpickle size.